Light Backends and Brokers Guides - PR from issue #6539

danthegoodman1 opened the pull request:
PR work for Issue: https://github.com/celery/celery/issues/6539

This adds some insight into when one might choose you use certain backends and brokers. This is not complete but the beginning of a framework for describing the advantages and disadvantages of each option.

matusvalo commented on README.rst:
```diff
@@ -213,6 +213,60 @@ database connections at ``fork``.
+RabbitMQ
+--------
+
+RabbitMQ is a broker.
```
I am missing here RabbitMQ as a backend usage using rpc:// backend

danthegoodman1 replied:
That goes outside my experience with Celery, I'll look into learning more about it though.

matusvalo reviewed (changes requested):
I prefer to have this kind of information in documentation instead of README.rst - e.g. in following file:

 celery/docs/getting-started/brokers/index.rst

matusvalo commented:
@danthegoodman1 you moved it to `docs/getting-started/index.rst`  instead of `celery/docs/getting-started/brokers/index.rst`. Me personal preference is to have it in brokers part of documentation instead of general one.

danthegoodman1 commented:
> @danthegoodman1 you moved it to `docs/getting-started/index.rst` instead of `celery/docs/getting-started/brokers/index.rst`. My personal preference is to have it in brokers part of documentation instead of general one.

@matusvalo I had assumed we might want to keep it out of the brokers folder since my changes also include lots of backend suggestions. Should I consider result stores brokers?

matusvalo commented:
Hmm good point. Let me review it once again :-)

matusvalo commented:
I have went over the documentation once again and still I prefer to have this information present in Celery 5.0.5 documentation » Getting Started » Brokers section. Maybe we can rename the whole section to Brokers and Backends.

danthegoodman1 commented:
Alright so I changed the name of the folder from `brokers` to `backends-and-brokers` and moved it into the `index.rst` there, along with a few language changes for modified page to make sense.

matusvalo commented:
Thank you @danthegoodman1. I merged your PR. I am still missing mentioning and explaining rpc backend but it can be done in another PR. If you have time any new PRs are welcome.

danthegoodman1 commented:
> Thank you @danthegoodman1. I merged your PR. I am still missing mentioning and explaining rpc backend but it can be done in another PR. If you have time any new PRs are welcome.

Thanks! I’ll definitely add more backends and brokers as I expand my experience with them.

matusvalo commented:
BTW I have created PR with rpc:// backend mentioned: #6568 . Feel free to review.
